File: python/cf.py
import http.client
import json

url = "codeforces.com"
options = '/api/contest.list'
link = 'http://codeforces.com/contest/'


def make_request(url, options, use_https=False):
    if use_https:
        conn = http.client.HTTPSConnection(url)
    else:
        conn = http.client.HTTPConnection(url)
    conn.request("GET", options)
    response = conn.getresponse()
    return response


response = make_request(url, options, use_https=True)

if response.status == 301 or response.status == 302:
    redirected_url = response.getheader('Location')
    if redirected_url:
        new_url = redirected_url.replace('http://', '').replace('https://', '')
        new_host, new_path = new_url.split('/', 1)
        new_path = '/' + new_path
        response = make_request(new_host, new_path, use_https=True)

if response.status == 200:
    data = json.loads(response.read().decode())

    div3 = []
    edu = []

    for fct in data['result']:
        if "Educational" in fct['name']:
            edu.append(fct['id'])
        if "Div. 3" in fct['name']:
            div3.append(fct['id'])

    div3.sort()
    edu.sort()

    with open("Div3.txt", "w") as div3_file:
        for i, contest_id in enumerate(div3, start=1):
            div3_file.write(f"Div3 Rounds {i} : {link}{contest_id}\n")

    with open("Educational.txt", "w") as edu_file:
        for i, contest_id in enumerate(edu, start=1):
            edu_file.write(f"Educational Round {i} : {link}{contest_id}\n")
else:
    print(f"Request failed with status code {response.status}")

File: node_js/cf.js
"use strict";

const fs = require("fs");
const url = "http://codeforces.com/api";
const options = "/contest.list";
const link = "http://codeforces.com/contest/";

let contest = async () => {
  let edu = [];
  let div3 = [];
  try {
    const response = await fetch(url + options);
    if (response.status == 200) {
      let data = await response.json();

      for (let i in data.result) {
        let fct = data.result[i];
        if (fct["name"].includes("Educational")) {
          edu.push(fct.id);
        }
        if (fct["name"].includes("Div. 3")) {
          div3.push(fct.id);
        }
      }
    } else {
      console.log(`Request failed with status code of ${response.status}`);
    }
  } catch (error) {
    console.log(
      "There may be some issues on Codeforces or the servers or not responding"
    );
    console.log("Please Wait and try it after few minutes");
    return;
  }

  div3.sort((a, b) => a - b);
  edu.sort((a, b) => a - b);

  for (let i in div3) {
    fs.appendFile(
      "Div3.txt",
      `Div3 Rounds ${Number(i) + 1} :  ${link + div3[i]}\n`,
      (err) => {
        if (err) console.log(err);
        else return;
      }
    );
  }

  for (let i in edu) {
    fs.appendFile(
      "Educational.txt",
      `Educational Round ${Number(i) + 1} :  ${link + edu[i]}\n`,
      (err) => {
        if (err) console.log(err);
        else return;
      }
    );
  }
};
contest();
